---
title: "Source Code - Pokorzynski & Carabeo 2021"
author: "Nick Pokorzynski"
date: "6/24/2021"
output: word_document
---

```{r}
#FIGURE 1
#Figure 1C
setwd("~/Documents/Carabeo Lab/R Code/16v24")
library(ggplot2)
library(ggpubr)

gdna<-read.csv("Ctr_gDNA_qPCR_05032021.csv")
gdna$Treatment2<-factor(gdna$Treatment, levels=c("Mock", "Bpd", "Trp"))
gdna$ID2<-factor(gdna$ID, levels=c("UTD24", "BPD16", "TRP16", "BPD24", "TRP24"))

gdnaaov<-aov(Value~ID, data=gdna)
summary(gdnaaov)
TukeyHSD(x=gdnaaov)

ggbarplot(gdna, x="ID2", y="Log10_Value", add="mean_sd", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3")) + 
     scale_y_continuous(expand=expand_scale(mult=c(0,0.1))) + 
     ylab(label="Log10 Genome Equivalents/ng gDNA") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") + 
     geom_exec(geomfunc=geom_point, data=gdna, x="ID2", y="Log10_Value", colour="black") + 
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("ns", "***", "***", "ns", "***", "***"), y_position=c(4.5, 5.5, 6, 4, 6.5, 7)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1))

#Figure 1D
setwd("~/Documents/Carabeo Lab/Zeiss Colibri7/IFU/16v24")
library(ggplot2)
library(ggpubr)

x<-read.csv("IFU_16v24_06012021.csv")
x$ID2<-factor(x$ID, levels=c("UTD24", "BPD16", "TRP16", "BPD24", "TRP24"))
x$Condition2<-factor(x$Condition, levels=c("UTD", "BPD", "TRP"))

ggbarplot(x, x="ID2", y="log_IFU_mL", add="mean_sd", fill="Condition2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3")) + 
     scale_y_continuous(expand=expand_scale(mult=c(0,0.1))) + 
     ylab(label="Log10 IFU/mL") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") + 
     geom_exec(geomfunc=geom_point, data=x, x="ID2", y="log_IFU_mL", colour="black") + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
  geom_hline(yintercept = 3.976781123, linetype = 2)

#Figure 1E
setwd("~/Documents/Carabeo Lab/R Code/16v24")
library(ggplot2)
library(ggpubr)
x<-read.csv("Ctr_RTqPCR_geom5_06152021.csv")
x$Gene2<-factor(x$Gene, levels=c("trpR", "trpB", "trpA", "ytgA", "omcB", "ompA"))
x$Treatment2<-factor(x$Treatment, levels=c("Mock", "Bpd", "Trp"))
x$ID2<-factor(x$ID, levels=c("UTD24", "BPD16", "TRP16", "BPD24", "TRP24"))

trpR<-subset(x, Gene%in%c("trpR"))
trpB<-subset(x, Gene%in%c("trpB"))
trpA<-subset(x, Gene%in%c("trpA"))
ytgA<-subset(x, Gene%in%c("ytgA"))
omcB<-subset(x, Gene%in%c("omcB"))
ompA<-subset(x, Gene%in%c("ompA"))

ggbarplot(x, x="ID2", y="log2", add=c("mean_sd", "point"), error.plot="upper_errorbar", color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), facet.by="Gene2", ncol=3) +
  scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(strip.text = element_text(face = "italic")) +
  geom_hline(yintercept=0.0) +
     stat_compare_means(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), label="p.signif", method="t.test", label.y=c(7, 9, 11, 7, 13, 15))

trpRaov<-aov(log2~ID, data=trpR)
summary(trpRaov)
TukeyHSD(x=trpRaov)
trpBaov<-aov(log2~ID, data=trpB)
summary(trpBaov)
TukeyHSD(x=trpBaov)
trpAaov<-aov(log2~ID, data=trpA)
summary(trpAaov)
TukeyHSD(x=trpAaov)
ytgAaov<-aov(log2~ID, data=ytgA)
summary(ytgAaov)
TukeyHSD(x=ytgAaov)
omcBaov<-aov(log2~ID, data=omcB)
summary(omcBaov)
TukeyHSD(x=omcBaov)
ompAaov<-aov(log2~ID, data=ompA)
summary(ompAaov)
TukeyHSD(x=ompAaov)

#single plots
ggbarplot(trpR, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "trpR") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("***", "***", "***", "***", "***", "***"), y_position = c(11.5, 7, 13, 11.5, 14, 15)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

ggbarplot(trpB, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "trpB") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("***", "***", "***", "***", "***", "***"), y_position = c(11.5, 7, 13, 11.5, 14, 15)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

ggbarplot(trpA, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "trpA") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("***", "***", "***", "***", "***", "***"), y_position = c(11.5, 7, 13, 11.5, 14, 15)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

ggbarplot(ytgA, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "ytgA") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("ns", "ns", "ns", "**", "ns", "ns"), y_position = c(11.5, 7, 13, 11.5, 14, 15)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

ggbarplot(omcB, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "omcB") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("***", "***", "***", "***", "***", "***"), y_position = c(9.5, 5, 11, 9.5, 13, 15)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

ggbarplot(ompA, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "ompA") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("***", "***", "***", "***", "***", "***"), y_position = c(9.5, 5, 11, 9.5, 13, 15)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)
```

```{r}
#FIGURE 6
#Figure 6B
setwd("~/Documents/Carabeo Lab/Mizoribine Treatments/gDNA")
library(ggplot2)
library(ggpubr)

gdna<-read.csv("MIZ_gDNA_qPCR_05102021.csv")
gdna$ID2<-factor(gdna$ID, levels=c("0", "12.5", "25", "50", "100", "200"))

gdnaaov<-aov(Value~ID2, data=gdna)
summary(gdnaaov)
TukeyHSD(x=gdnaaov)

ggbarplot(gdna, x="ID2", y="Log10_Value", add=c("mean_sd", "point"), fill="ID2", palette=c("white", "darkseagreen1", "darkseagreen2", "darkseagreen3", "darkseagreen", "darkseagreen4")) + 
     scale_y_continuous(expand=expand_scale(mult=c(0,0.1))) + 
     ylab(label="Log10 Genome Equivalents/ng gDNA") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     xlab("Mizoribine Concentration (µM)") +
     geom_signif(comparisons=list(c("0", "12.5"), c("12.5", "25"), c("25", "50"), c("50", "100"), c("100", "200")), annotation=c("***", "***", "***", "ns", "ns"), y_position=c(6,6,6,6,6)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1))

#Figure 6C
setwd("~/Documents/Carabeo Lab/Mizoribine Treatments/IFU")
library(ggplot2)
library(ggpubr)

x<-read.csv("MIZ_IFU_05192021.csv")
x$ID2<-factor(x$ID, levels=c("UTD24", "MIZ24", "UTD24+16", "MIZ24+16"))
x$Condition2<-factor(x$Condition, levels=c("UTD", "MIZ"))

ggbarplot(x, x="ID2", y="log_IFU_mL", add=c("mean_sd", "point"), fill="Condition2", palette=c("white", "darkseagreen")) + 
     scale_y_continuous(expand=expansion(mult=c(0,0.1))) + 
     ylab(label="Log10 IFU/mL") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
  geom_hline(yintercept = 3.976781123, linetype = 2)

#Figure 6D
setwd("~/Documents/Carabeo Lab/Mizoribine Treatments/RT-qPCR")
library(ggplot2)
library(ggpubr)
x<-read.csv("MIZ_geom5_RTqPCR_06152021.csv")
x$Gene2<-factor(x$Gene, levels=c("trpR", "trpB", "trpA", "ytgA", "omcB", "ompA"))
x$ID2<-factor(x$ID, levels=c("UTD24", "MIZ24"))

trpR<-subset(x, Gene%in%c("trpR"))
trpB<-subset(x, Gene%in%c("trpB"))
trpA<-subset(x, Gene%in%c("trpA"))
ytgA<-subset(x, Gene%in%c("ytgA"))
omcB<-subset(x, Gene%in%c("omcB"))
ompA<-subset(x, Gene%in%c("ompA"))

t.test(log2 ~ ID, data = trpR)
t.test(log2 ~ ID, data = trpB)
t.test(log2 ~ ID, data = trpA)
t.test(log2 ~ ID, data = ytgA)
t.test(log2 ~ ID, data = omcB)
t.test(log2 ~ ID, data = ompA)


ggbarplot(x, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="ID2", palette=c("white", "chartreuse3"), facet.by="Gene2", ncol=3) +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(strip.text = element_text(face = "italic")) +
     stat_compare_means(comparisons=list(c("UTD24", "MIZ24")), label="p.signif", method="t.test", label.y=c(3.5)) +
  geom_hline(yintercept = 0.0)
```

```{r}
#FIGURE 7
#Figure 7B
setwd("~/Documents/Carabeo Lab/GTPase-Glo")
library(ggpubr)
library(ggplot2)
x<-read.csv("GTP_05072021_3.csv")

xaov<-aov(Normalized_Value~Treatment+Condition+Treatment:Condition, data=x)
summary(xaov)
TukeyHSD(x=xaov)

x$Treatment2<-factor(x$Treatment, levels=c("UTD24", "MIZ24", "BPD24", "TRP24"))
x$Condition2<-factor(x$Condition, levels=c("Mock", "Ctr"))

ggbarplot(x, x="Condition2", y="Normalized_Value", add=c("mean_sd", "point"), color="black", fill="Condition2", palette=c("darkolivegreen1", "darkolivegreen"), position = position_dodge(), facet.by = "Treatment2", ncol=4) +
     scale_y_continuous(expand=expand_scale(mult=c(0,0.1))) +
     ylab(label = "Normalized GTP Levels\n(Luminescence)") +
     rremove("xlab") +
     rremove("legend") +
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
     stat_compare_means(comparisons = list(c("Mock", "Ctr")), label="p.signif", method="t.test", label.y=c(650000))

#Figure 7C
setwd("~/Documents/Carabeo Lab/R Code/16v24")
library(ggplot2)
library(ggpubr)

x<-read.csv("Ctr_RTqPCR_cofactor_geom5_06152021.csv")
x$Gene2<-factor(x$Gene, levels=c("ribA", "ribF", "trpC", "ctl0875"))
x$Treatment2<-factor(x$Treatment, levels=c("Mock", "Bpd", "Trp"))
x$ID2<-factor(x$ID, levels=c("UTD24", "BPD16", "TRP16", "BPD24", "TRP24"))

ribA<-subset(x, Gene%in%c("ribA"))
trpC<-subset(x, Gene%in%c("trpC"))
ribF<-subset(x, Gene%in%c("ribF"))
ctl0875<-subset(x, Gene%in%c("ctl0875"))

ribAaov<-aov(log2~ID, data=ribA)
summary(ribAaov)
TukeyHSD(x=ribAaov)
trpCaov<-aov(log2~ID, data=trpC)
summary(trpCaov)
TukeyHSD(x=trpCaov)
ribFaov<-aov(log2~ID, data=ribF)
summary(ribFaov)
TukeyHSD(x=ribFaov)
ctl0875aov<-aov(log2~ID, data=ctl0875)
summary(ctl0875aov)
TukeyHSD(x=ctl0875aov)

ggbarplot(ribA, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "ribA") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("***", "*", "***", "ns", "***", "***"), y_position = c(1.5, 2, 2.5, 1.5, 3, 3.5)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

ggbarplot(trpC, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "trpC") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("ns", "***", "***", "*", "ns", "***"), y_position = c(1.5, 2, 2.5, 1.5, 3, 3.5)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

ggbarplot(ribF, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "ribF") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("ns", "ns", "ns", "*", "***", "*"), y_position = c(1.5, 2, 2.5, 1.5, 3, 3.5)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

ggbarplot(ctl0875, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "ctl0875") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("*", "ns", "ns", "***", "***", "ns"), y_position = c(1.5, 2, 2.5, 1.5, 3, 3.5)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

#Figure 7D
setwd("~/Documents/Carabeo Lab/Mizoribine Treatments/RT-qPCR")
library(ggplot2)
library(ggpubr)
x<-read.csv("MIZ_geom5_cofactor_RTqPCR_06152021.csv")
x$Gene2<-factor(x$Gene, levels=c("ribA", "ribF", "trpC", "ctl0875"))
x$ID2<-factor(x$ID, levels=c("UTD24", "MIZ24"))

ribA<-subset(x, Gene%in%c("ribA"))
ribF<-subset(x, Gene%in%c("ribF"))
trpC<-subset(x, Gene%in%c("trpC"))
ctl0875<-subset(x, Gene%in%c("ctl0875"))

t.test(log2 ~ ID, data = ribA)
t.test(log2 ~ ID, data = ribF)
t.test(log2 ~ ID, data = trpC)
t.test(log2 ~ ID, data = ctl0875)


ggbarplot(x, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="ID2", palette=c("white", "chartreuse3"), facet.by="Gene2", ncol=4) +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(strip.text = element_text(face = "italic")) +
     stat_compare_means(comparisons=list(c("UTD24", "MIZ24")), label="p.signif", method="t.test", label.y=c(3.5)) +
  geom_hline(yintercept = 0.0)
```

```{r}
#FIGURE S1
setwd("~/Documents/Carabeo Lab/R Code/16v24")
library(ggplot2)
library(ggpubr)
library(SLqPCR)
library(tibble)

geNorm<-read.csv("geNorm4.csv")
geNorm<-column_to_rownames(geNorm, 'X')
res<-selectHKgenes(geNorm, method = "Vandesompele", geneSymbol = names(geNorm), minNrHK = 2, trace = TRUE, na.rm = TRUE)

library(RColorBrewer)
mypalette <- brewer.pal(5, "Set1")
matplot(res$meanM, type = "b", ylab = "Average expression stability M", xlab = "Number of remaining control genes", axes = FALSE, pch = 19, col = mypalette, ylim = c(0.02, 0.12), lty = 1, lwd = 2, main = "Gene stability measure")
axis(1, at = 1:6, labels = as.character(7:2))
axis(2, at = seq(0.02, 0.12, by = 0.02), labels = as.character(seq(0.02, 0.12, by = 0.02)))
box()
abline(h = seq(0.02, 0.12, by = 0.02), lty = 2, lwd = 1, col = "grey")

mypalette <- brewer.pal(9, "YlGnBu")
barplot(res$variation, beside = TRUE, col = mypalette, names.arg = "Stepwise Iterations", ylim=c(0, 0.15), ylab = "Pairwise Variation")
legend("topright", legend = c("V6/7", "V5/6", "V4/5", "V3/4", "V2/3"), fill = mypalette, ncol = 2)
abline(h = seq(0.02, 0.15, by = 0.02), lty = 2, col = "grey")
box()

write.csv(as.data.frame(res$ranking), file="geNorm4_Ranking_06152021.csv")
write.csv(as.data.frame(res$variation), file="geNorm4_Variation_06152021.csv")
write.csv(as.data.frame(res$meanM), file="geNorm4_meanM_06152021.csv")
```


```{r}
#FIGURE S2
setwd("~/Documents/Carabeo Lab/R Code/16v24")
library(ggplot2)
library(ggpubr)
x<-read.csv("Ctr_RTqPCR_unique_geom5_06152021.csv")
x$Gene2<-factor(x$Gene, levels=c("amiA", "atpB", "dapL", "tyrP"))
x$Treatment2<-factor(x$Treatment, levels=c("Mock", "Bpd", "Trp"))
x$ID2<-factor(x$ID, levels=c("UTD24", "BPD16", "TRP16", "BPD24", "TRP24"))

amiA<-subset(x, Gene%in%c("amiA"))
atpB<-subset(x, Gene%in%c("atpB"))
dapL<-subset(x, Gene%in%c("dapL"))
tyrP<-subset(x, Gene%in%c("tyrP"))

amiAaov<-aov(log2~ID, data=amiA)
summary(amiAaov)
TukeyHSD(x=amiAaov)
atpBaov<-aov(log2~ID, data=atpB)
summary(atpBaov)
TukeyHSD(x=atpBaov)
dapLaov<-aov(log2~ID, data=dapL)
summary(dapLaov)
TukeyHSD(x=dapLaov)
tyrPaov<-aov(log2~ID, data=tyrP)
summary(tyrPaov)
TukeyHSD(x=tyrPaov)

ggbarplot(amiA, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "amiA") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("ns", "ns", "**", "***", "***", "ns"), y_position = c(1.5, 2, 2.5, 1.5, 3, 3.5)) +
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

ggbarplot(atpB, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "atpB") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("***", "ns", "***", "ns", "***", "***"), y_position = c(1.5, 2, 2.5, 1.5, 3, 3.5)) +
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

ggbarplot(dapL, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "dapL") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("***", "***", "***", "***", "***", "***"), y_position = c(1.75, 2, 2.5, 1.75, 3, 3.5)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)

ggbarplot(tyrP, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="Treatment2", palette=c("lightsteelblue4", "lightsteelblue1", "lightsteelblue3"), title = "tyrP") +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(plot.title = element_text(face = "italic")) +
     geom_signif(comparisons=list(c("BPD16", "TRP16"), c("UTD24", "BPD16"), c("UTD24", "TRP16"), c("BPD24", "TRP24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("***", "**", "***", "***", "***", "***"), y_position = c(1.75, 2, 2.5, 1.75, 3, 3.5)) + 
     theme(line=element_line(size=1, colour="black"), panel.border=element_rect(colour="black", fill="NA", size=1)) +
    theme(plot.title = element_text(hjust=0.5)) +
  geom_hline(yintercept=0.0)
```

```{r}
#FIGURE S4
setwd("~/Documents/Carabeo Lab/Mizoribine Treatments/RT-qPCR")
library(ggplot2)
library(ggpubr)
x<-read.csv("MIZ_geom5_unique_RTqPCR_06152021.csv")
x$Gene2<-factor(x$Gene, levels=c("amiA", "atpB", "dapL", "tyrP"))
x$ID2<-factor(x$ID, levels=c("UTD24", "MIZ24"))

amiA<-subset(x, Gene%in%c("amiA"))
atpB<-subset(x, Gene%in%c("atpB"))
dapL<-subset(x, Gene%in%c("dapL"))
tyrP<-subset(x, Gene%in%c("tyrP"))

t.test(log2 ~ ID, data = amiA)
t.test(log2 ~ ID, data = atpB)
t.test(log2 ~ ID, data = dapL)
t.test(log2 ~ ID, data = tyrP)


ggbarplot(x, x="ID2", y="log2", add=c("mean_sd", "point"), color="black", fill="ID2", palette=c("white", "chartreuse3"), facet.by="Gene2", ncol=2) +  
     scale_y_continuous(expand=expansion(mult=c(0.1, 0.1))) + 
     ylab(label="Log2 Transcript Expression") + 
     rremove("legend") + 
     theme(axis.text.x=element_text(angle=45, hjust=1)) + 
     rremove("xlab") +
     theme(strip.text = element_text(face = "italic")) +
     stat_compare_means(comparisons=list(c("UTD24", "MIZ24")), label="p.signif", method="t.test", label.y=c(3.5)) +
  geom_hline(yintercept = 0.0)
```

```{r}
#FIGURE S5
setwd("~/Documents/Carabeo Lab/GTPase-Glo")
library(ggpubr)
library(ggplot2)

x<-read.csv("GTP_raw_05102021.csv")
x$Measurement2<-factor(x$Measurement, levels=c("[ATP]", "[GTP]+[ATP]"))
x$Treatment2<-factor(x$Treatment, levels=c("UTD24", "MIZ24", "BPD24", "TRP24"))
x$Condition2<-factor(x$Condition, levels=c("Mock", "Ctr"))

ATP<-subset(x, Measurement%in%c("[ATP]"))
GTP<-subset(x, Measurement%in%c("[GTP]+[ATP]"))

atpaov<-aov(Value~Treatment, data=ATP)
summary(atpaov)
TukeyHSD(x=atpaov)
gtpaov<-aov(Value~Treatment, data = GTP)
summary(gtpaov)
TukeyHSD(x=gtpaov)

#Figure S4A
ggbarplot(ATP, "Treatment2", "Value", add=c("mean_sd", "point"), color="black", fill="Condition2", facet.by = "Condition2", ncol=2, title = "[ATP]", palette = c("olivedrab1", "olivedrab")) +
  rremove("legend") +
  theme(axis.title.x = element_blank()) +
     scale_y_continuous(expand=expand_scale(mult=c(0,0.1)), limits = c(0, 6e6)) +
     theme(plot.title = element_text(hjust = '0.5')) +
     ylab("Luminescence") +
     theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  geom_signif(comparisons=list(c("UTD24", "MIZ24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("ns", "***", "ns"), y_position=c(4500000,5000000,5500000))

#Figure S4B
ggbarplot(GTP, "Treatment2", "Value", add=c("mean_sd", "point"), color="black", fill="Condition2", facet.by = "Condition2", ncol=2, title = "[GTP]+[ATP]", palette = c("olivedrab1", "olivedrab")) +
  rremove("legend") +
  theme(axis.title.x = element_blank()) +
     scale_y_continuous(expand=expand_scale(mult=c(0,0.1)), limits = c(0, 6.5e6)) +
     theme(plot.title = element_text(hjust = '0.5')) +
     ylab("Luminescence") +
     theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  geom_signif(comparisons=list(c("UTD24", "MIZ24"), c("UTD24", "BPD24"), c("UTD24", "TRP24")), annotation=c("ns", "***", "ns"), y_position=c(5500000,6000000,6500000))
```

```{r}
#DESeq2 - Chlamydia trachomatis
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/Ctr/FINAL")
#install shiny packages for bioconductor, make sure to include DESeq2, tibble, plyr, dplyr, ggplot2, ggfortify
library(DESeq2)
library(tibble)
library(plyr)
library(dplyr)
library(ggplot2)
library(ggfortify)
library(EnhancedVolcano)
#Chlamydia DESeq2
#From htseq-counts file generated in Galaxy main (usegalaxy.org), compile csv with all counts for each conditions
cts<-read.csv("Ctr_htseq_04082021.csv")
cts2<-cts%>%column_to_rownames("X")

#generate coldata file to specify sample information
coldata<-read.csv("Ctr_coldata_04082021.csv")
coldata2<-coldata%>%column_to_rownames("sample")

#generate dds object
dds<-DESeqDataSetFromMatrix(countData=cts2, colData=coldata2, design=~condition)

#filter genes with less than 10 read counts
keep<-rowSums(counts(dds))>=10
dds<-dds[keep,]

#set reference level
dds$condition<-relevel(dds$condition, ref="UTD")

#run DESeq2
dds<-DESeq(dds)
resultsNames(dds)

#extract tables for each comparison
resBPD16<-lfcShrink(dds, coef=2, type='apeglm')
resBPD24<-lfcShrink(dds, coef=3, type='apeglm')
resTRP16<-lfcShrink(dds, coef=4, type='apeglm')
resTRP24<-lfcShrink(dds, coef=5, type='apeglm')

#save each table to csv
write.csv(as.data.frame(resBPD16), file="Ctr_BPD16_DESeq2_04082021.csv")
write.csv(as.data.frame(resBPD24), file="Ctr_BPD24_DESeq2_04082021.csv")
write.csv(as.data.frame(resTRP16), file="Ctr_TRP16_DESeq2_04082021.csv")
write.csv(as.data.frame(resTRP24), file="Ctr_TRP24_DESeq2_04082021.csv")

#rlog transform data for PCA
rld<-rlog(dds, blind=FALSE)

#plot pca using plotPCA and ggplot functions
pca<-plotPCA(rld, intgroup="condition", returnData=TRUE)
pVar<-round(100*attr(pca, "percentVar"))
ggplot(pca, aes(PC1, PC2, color=condition)) +
    geom_point(size=3) +
    xlab(paste0("PC1: ",pVar[1],"% variance")) +
    ylab(paste0("PC2: ",pVar[2],"% variance")) +
    coord_fixed() +
    theme_linedraw() +
    theme(legend.title=element_blank()) +
    ggtitle("C. trachomatis PCA") +
    theme(plot.title = element_text(hjust=0.5)) +
    geom_text(aes(label = name), hjust=0, vjust=0) +
    xlim(-30,20)

#plot MA plots using plotMA
plotMA(resBPD16, main="BPD16 \n C. trachomatis")
plotMA(resBPD24, main="BPD24 \n C. trachomatis")
plotMA(resTRP16, main="TRP16 \n C. trachomatis")
plotMA(resTRP24, main="TRP24 \n C. trachomatis")

#extract Normalized counts for gage
dds <- estimateSizeFactors(dds)
ncts<-counts(dds, normalized=TRUE)
write.csv(ncts, file="Ctr_NormCounts_DESeq2_12012020.csv")

#EnhancedVolcano Plots
#csv files were annotated with symbols, etc. manually through the Galaxy server to provide the information for the plots
#BPD16
BPD16_vol<-read.csv("Ctr_BPD16_DESeq2_04082021.csv")
EnhancedVolcano(BPD16_vol, lab=BPD16_vol$Symbol, x='log2FoldChange', y='padj', FCcutoff = 0.58496250072, pCutoff = 5e-2, legendPosition = '', title="BPD16", subtitle="Reference = UTD24") + theme(plot.title = element_text(hjust = 0.5)) + theme(plot.subtitle = element_text(hjust = 0.5))

#TRP16
TRP16_vol<-read.csv("Ctr_TRP16_DESeq2_04082021.csv")
EnhancedVolcano(TRP16_vol, lab=TRP16_vol$Symbol, x='log2FoldChange', y='padj', FCcutoff = 0.58496250072, pCutoff = 5e-2, legendPosition = '', title="TRP16", subtitle="Reference = UTD24") + theme(plot.title = element_text(hjust = 0.5)) + theme(plot.subtitle = element_text(hjust = 0.5))

#BPD24
BPD24_vol<-read.csv("Ctr_BPD24_DESeq2_04082021.csv")
EnhancedVolcano(BPD24_vol, lab=BPD24_vol$Symbol, x='log2FoldChange', y='padj', FCcutoff = 0.58496250072, pCutoff = 5e-2, legendPosition = '', title="BPD24", subtitle="Reference = UTD24") + theme(plot.title = element_text(hjust = 0.5)) + theme(plot.subtitle = element_text(hjust = 0.5))

#TRP24
TRP24_vol<-read.csv("Ctr_TRP24_DESeq2_04082021.csv")
EnhancedVolcano(TRP24_vol, lab=TRP24_vol$Symbol, x='log2FoldChange', y='padj', FCcutoff = 0.58496250072, pCutoff = 5e-2, legendPosition = '', title="TRP24", subtitle="Reference = UTD24") + theme(plot.title = element_text(hjust = 0.5)) + theme(plot.subtitle = element_text(hjust = 0.5))

#clusterProfiler analysis
 if (!requireNamespace("BiocManager", quietly = TRUE))
+     install.packages("BiocManager")
 
BiocManager::install("clusterProfiler")
install.packages('ggnewscale')
install.packages('ggridges')
install.packages('clusterProfiler.dplyr')
library(ggnewscale)
library(ggridges)
library(DOSE)
library(ggplot2)
library(clusterProfiler)
library(dplyr)
library(org.Hs.eg.db)
library(enrichplot)
library(GOSemSim)
library(pathview)
#may need to update ggrepel, ggplot2, geforce, ggraph packages to get cnetplot and emapplot to work
#BPD16
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/Ctr/FINAL")
#read data from DESeq2 output csv that has been annotated already
BPD16<-read.csv("Ctr_BPD16_DESeq2_04082021.csv")

#select log2foldchange data
BPD16list<-BPD16[,3]
#construct vector with ENSEMBL gene IDs to define fold change values
names(BPD16list)<-as.character(BPD16[,1])
#sort as decreasing list
BPD16list<-sort(BPD16list, decreasing=TRUE)
#perform gene set enrichment analysis using KEGG pathway enrichment
BPD16k <- gseKEGG(geneList = BPD16list, organism = 'ctb', minGSSize = 20, pvalueCutoff = 0.05, verbose = FALSE)
write.csv(as.data.frame(BPD16k), file="Ctr_BPD16_gseKEGG_04092021.csv")

#dotplot of enriched KEGG terms
dotplot(BPD16k, showCategory = 5, orderBy = "x") +
  ggtitle("BPD16 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_x_continuous(limits=c(0.4, 1.0))

#ridgeplot of enriched KEGG terms by gene expression
ridgeplot(BPD16k, showCategory = 5) +
  ggtitle("BPD16 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab("Log2 Fold Change")

#plot pathview for enriched KEGG terms
BPD16l<-read.csv("Ctr_BPD16_gseKEGG_04092021.csv")
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/Ctr/FINAL/clusterProfiler/BPD16")
tmp = sapply(BPD16l, function(pid) pathview(gene.data=BPD16list, gene.idtype="kegg", pathway.id=pid, species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#riboflavin metabolism
tmp = sapply(BPD16l, function(pid) pathview(gene.data=BPD16list, gene.idtype="kegg", pathway.id='ctb00740', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#folate biosynthesis
tmp = sapply(BPD16l, function(pid) pathview(gene.data=BPD16list, gene.idtype="kegg", pathway.id='ctb00790', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#porphyrin and chlorophyll metabolism
tmp = sapply(BPD16l, function(pid) pathview(gene.data=BPD16list, gene.idtype="kegg", pathway.id='ctb00860', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))

#TRP16
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/Ctr/FINAL")
TRP16<-read.csv("Ctr_TRP16_DESeq2_04082021.csv")

TRP16list<-TRP16[,3]
names(TRP16list)<-as.character(TRP16[,1])
TRP16list<-sort(TRP16list, decreasing=TRUE)
TRP16k <- gseKEGG(geneList = TRP16list, organism = 'ctb', minGSSize = 20, pvalueCutoff = 0.05, verbose = FALSE)
write.csv(as.data.frame(TRP16k), file="Ctr_TRP16_gseKEGG_04092021.csv")

dotplot(TRP16k, showCategory = 5, orderBy = "x") +
  ggtitle("TRP16 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_x_continuous(limits=c(0.2, 1.0))

ridgeplot(TRP16k, showCategory = 5) +
  ggtitle("TRP16 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab("Log2 Fold Change")

TRP16l<-read.csv("Ctr_TRP16_gseKEGG_04092021.csv")
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/Ctr/FINAL/clusterProfiler/TRP16")
tmp = sapply(TRP16l, function(pid) pathview(gene.data=TRP16list, gene.idtype="kegg", pathway.id=pid, species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#riboflavin metabolism
tmp = sapply(TRP16l, function(pid) pathview(gene.data=TRP16list, gene.idtype="kegg", pathway.id='ctb00740', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#folate biosynthesis
tmp = sapply(TRP16l, function(pid) pathview(gene.data=TRP16list, gene.idtype="kegg", pathway.id='ctb00790', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#porphyrin and chlorophyll metabolism
tmp = sapply(TRP16l, function(pid) pathview(gene.data=TRP16list, gene.idtype="kegg", pathway.id='ctb00860', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))

#BPD24
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/Ctr/FINAL")
BPD24<-read.csv("Ctr_BPD24_DESeq2_04082021.csv")

BPD24list<-BPD24[,3]
names(BPD24list)<-as.character(BPD24[,1])
BPD24list<-sort(BPD24list, decreasing=TRUE)
BPD24k <- gseKEGG(geneList = BPD24list, organism = 'ctb', minGSSize = 20, pvalueCutoff = 0.05, verbose = FALSE)
write.csv(as.data.frame(BPD24k), file="Ctr_BPD24_gseKEGG_04092021.csv")

dotplot(BPD24k, showCategory = 5, orderBy = "x") +
  ggtitle("BPD24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_x_continuous(limits=c(0.4, 1.0))

ridgeplot(BPD24k, showCategory = 5) +
  ggtitle("BPD24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab("Log2 Fold Change")

BPD24l<-read.csv("Ctr_BPD24_gseKEGG_04092021.csv")
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/Ctr/FINAL/clusterProfiler/BPD24")
tmp = sapply(BPD24l, function(pid) pathview(gene.data=BPD24list, gene.idtype="kegg", pathway.id=pid, species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#riboflavin metabolism
tmp = sapply(BPD24l, function(pid) pathview(gene.data=BPD24list, gene.idtype="kegg", pathway.id='ctb00740', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#folate biosynthesis
tmp = sapply(BPD24l, function(pid) pathview(gene.data=BPD24list, gene.idtype="kegg", pathway.id='ctb00790', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#porphyrin and chlorophyll metabolism
tmp = sapply(BPD24l, function(pid) pathview(gene.data=BPD24list, gene.idtype="kegg", pathway.id='ctb00860', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#lysine biosynthesis
tmp = sapply(BPD24l, function(pid) pathview(gene.data=BPD24list, gene.idtype="kegg", pathway.id='ctb00300', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#peptidoglycan biosynthesis
tmp = sapply(BPD24l, function(pid) pathview(gene.data=BPD24list, gene.idtype="kegg", pathway.id='ctb00550', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))

#TRP24
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/Ctr/FINAL")

TRP24<-read.csv("Ctr_TRP24_DESeq2_04082021.csv")
TRP24list<-TRP24[,3]
names(TRP24list)<-as.character(TRP24[,1])
TRP24list<-sort(TRP24list, decreasing=TRUE)
TRP24k <- gseKEGG(geneList = TRP24list, organism = 'ctb', minGSSize = 20, pvalueCutoff = 0.05, verbose = FALSE)
write.csv(as.data.frame(TRP24k), file="Ctr_TRP24_gseKEGG_04092021.csv")

dotplot(TRP24k, showCategory = 5, orderBy = "x") +
  ggtitle("TRP24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_x_continuous(limits=c(0.2, 1.0))

ridgeplot(TRP24k, showCategory = 5) +
  ggtitle("TRP24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab("Log2 Fold Change")

TRP24l<-read.csv("Ctr_TRP24_gseKEGG_04092021.csv")
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/Ctr/FINAL/clusterProfiler/TRP24")
tmp = sapply(TRP24l, function(pid) pathview(gene.data=TRP24list, gene.idtype="kegg", pathway.id=pid, species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#riboflavin metabolism
tmp = sapply(TRP24l, function(pid) pathview(gene.data=TRP24list, gene.idtype="kegg", pathway.id='ctb00740', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#folate biosynthesis
tmp = sapply(TRP24l, function(pid) pathview(gene.data=TRP24list, gene.idtype="kegg", pathway.id='ctb00790', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))
#porphyrin and chlorophyll metabolism
tmp = sapply(TRP24l, function(pid) pathview(gene.data=TRP24list, gene.idtype="kegg", pathway.id='ctb00860', species="ctb", same.layer = F, low="purple", mid="gray", high="yellow"))

#To generate lists of matched up- or down-regualted genes between conditions, I defined up- or down-regulated sets using the filter function in Excel and then re-imported csv files of each dataset for each condition. Then I used the plyr/dplyr "match_df" function to generate lists of matched genes between conditions
#import csvs
bpd24up<-read.csv("bpd24up.csv")
trp24up<-read.csv("trp24up.csv")
#match on ID term
up<-match_df(trp24up, bpd24up, on="Symbol")
#write list to csv file
write.csv(up, file="Ctr_up24_matchdf_04102021.csv")
#repeat for each contrast of interest
bpd24down<-read.csv("bpd24down.csv")
trp24down<-read.csv("trp24down.csv")
down<-match_df(trp24down, bpd24down, on="Symbol")
write.csv(down, file="Ctr_down24_matchdf_04102021.csv")
bpd16down<-read.csv("bpd16down.csv")
trp16down<-read.csv("trp16down.csv")
down<-match_df(trp16down, bpd16down, on="Symbol")
write.csv(down, file="Ctr_down16_matchdf_04102021.csv")
bpd16up<-read.csv("bpd16up.csv")
trp16up<-read.csv("trp16up.csv")
up<-match_df(trp16up, bpd16up, on="Symbol")
write.csv(up, file="Ctr_up16_matchdf_04102021.csv")
up<-match_df(bpd16up, bpd24up, on="Symbol")
write.csv(up, file="Ctr_bpdup_matchdf_04102021.csv")
down<-match_df(bpd16down, bpd24down, on="Symbol")
write.csv(down, file="Ctr_bpddown_matchdf_04102021.csv")
up<-match_df(trp16up, trp24up, on="Symbol")
write.csv(up, file="Ctr_trpup_matchdf_04102021.csv")
down<-match_df(trp16down, trp24down, on="Symbol")
write.csv(down, file="Ctr_trpdown_matchdf_04102021.csv")

#Use of "dupsBetweenGroups" function for identifying unqique elements in a dataframe
#data frames must be organized with rows corresponding to condition-dependent values, e.g. row 1 = UTD24, value, value, etc.
#define function (obtained from http://www.cookbook-r.com/Manipulating_data/Comparing_data_frames/):
dupsBetweenGroups <- function (df, idcol) {
    # df: the data frame
    # idcol: the column which identifies the group each row belongs to

    # Get the data columns to use for finding matches
    datacols <- setdiff(names(df), idcol)

    # Sort by idcol, then datacols. Save order so we can undo the sorting later.
    sortorder <- do.call(order, df)
    df <- df[sortorder,]

    # Find duplicates within each id group (first copy not marked)
    dupWithin <- duplicated(df)

    # With duplicates within each group filtered out, find duplicates between groups. 
    # Need to scan up and down with duplicated() because first copy is not marked.
    dupBetween = rep(NA, nrow(df))
    dupBetween[!dupWithin] <- duplicated(df[!dupWithin,datacols])
    dupBetween[!dupWithin] <- duplicated(df[!dupWithin,datacols], fromLast=TRUE) | dupBetween[!dupWithin]

    # ============= Replace NA's with previous non-NA value ==============
    # This is why we sorted earlier - it was necessary to do this part efficiently

    # Get indexes of non-NA's
    goodIdx <- !is.na(dupBetween)

    # These are the non-NA values from x only
    # Add a leading NA for later use when we index into this vector
    goodVals <- c(NA, dupBetween[goodIdx])

    # Fill the indices of the output vector with the indices pulled from
    # these offsets of goodVals. Add 1 to avoid indexing to zero.
    fillIdx <- cumsum(goodIdx)+1

    # The original vector, now with gaps filled
    dupBetween <- goodVals[fillIdx]

    # Undo the original sort
    dupBetween[sortorder] <- dupBetween

    # Return the vector of which entries are duplicated across groups
    return(dupBetween)
}

#read in files with up- or down-regulated genes matched by condition ID in separate columns i.e. two columns, one specifying ID and the other specifying Gene name 
ctr_up<-read.csv("ctr_up_unique.csv")
dupRows<-dupsBetweenGroups(ctr_up, "ID")
ctr_up_unique<-cbind(ctr_up, unique=!dupRows)
write.csv(ctr_up_unique, file="Ctr_unique_up_04102021.csv")
ctr_down<-read.csv("ctr_down_unique.csv")
dupRows<-dupsBetweenGroups(ctr_down, "ID")
ctr_down_unique<-cbind(ctr_down, unique=!dupRows)
write.csv(ctr_down_unique, file="Ctr_unique_down_04102021.csv")
ctr_up<-read.csv("ctr_up_unique_all.csv")
dupRows<-dupsBetweenGroups(ctr_up, "ID")
ctr_up_unique<-cbind(ctr_up, unique=!dupRows)
write.csv(ctr_up_unique, file="Ctr_unique_up_all_04102021.csv")
ctr_down<-read.csv("ctr_down_unique_all.csv")
dupRows<-dupsBetweenGroups(ctr_down, "ID")
ctr_down_unique<-cbind(ctr_down, unique=!dupRows)
write.csv(ctr_down_unique, file="Ctr_unique_down_all_04102021.csv")

#to generate Venn diagram plots, I used the package ggVennDiagram, which is very simple and straight forward
#data frames were generated in csv and imported that simply listed each set of up- or down-regulated genes for each condition and then input to ggvenndiagram
#the only issue with ggvenndiagram is that I havent yet figured out a way to populate VD plots with comparisons that skip a column in the data set, i.e. I can compare columns 1 and 2 and columns 3 and 4 (or all columns) but not 1 and 3 or 2 and 4, etc. 
#load ggVennDiagram package
library(ggVennDiagram)
#import dataset
y<-read.csv("Ctr_venn_DOWN_04102021.csv")
#generate VD plot with all conditions
ggVennDiagram(y) + ggtitle("C. trachomatis\n Down-regulated") + theme(plot.title=element_text(hjust=0.5))
#generate VD plot with just 16h treatment conditions
ggVennDiagram(y[1:2]) + ggtitle("C. trachomatis\n 16h Down-regulated") + theme(plot.title=element_text(hjust=0.5))
#generate VD plot with just 24h treatment conditions
ggVennDiagram(y[3:4]) + ggtitle("C. trachomatis\n 24h Down-regulated") + theme(plot.title=element_text(hjust=0.5))
#repeat for up-regulated genes
x<-read.csv("Ctr_venn_UP_04102021.csv")
ggVennDiagram(x) + ggtitle("C. trachomatis\n Up-regulated") + theme(plot.title=element_text(hjust=0.5))
ggVennDiagram(x[1:2]) + ggtitle("C. trachomatis\n 16h Up-regulated") + theme(plot.title=element_text(hjust=0.5))
ggVennDiagram(x[3:4]) + ggtitle("C. trachomatis\n 24h Up-regulated") + theme(plot.title=element_text(hjust=0.5))
```

```{r}
#DESeq2 - HeLa 
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/HeLa/Final")
cts<-read.csv("hela_htseq_04042021.csv")
cts2<-cts%>%column_to_rownames("X")
coldata<-read.csv("hela_coldata_04042021.csv")
coldata2<-coldata%>%column_to_rownames("sample")
dds<-DESeqDataSetFromMatrix(countData=cts2, colData=coldata2, design=~condition)
keep<-rowSums(counts(dds))>=10
dds<-dds[keep,]
dds$condition<-relevel(dds$condition, ref="UTD")
#repeat analysis with ref="Mock" to get UTD24 results normalized to Mock infected group
dds<-DESeq(dds)
resultsNames(dds)

#perform lfcShrink on data using coef equal to resultsNames identity
resBPD16<-lfcShrink(dds, coef=2, type='apeglm')
resBPD24<-lfcShrink(dds, coef=3, type='apeglm')
resTRP16<-lfcShrink(dds, coef=5, type='apeglm')
resTRP24<-lfcShrink(dds, coef=6, type='apeglm')
#Note that Mock v UTD will show DE relative to UTD24!
resMock<-lfcShrink(dds, coef=4, type='apeglm')

#annotate files
library(AnnotationDbi)
library(org.Hs.eg.db)

#BPD16
resBPD16$symbol<-mapIds(org.Hs.eg.db, keys=row.names(resBPD16), column="SYMBOL", keytype="ENSEMBL", multiVales="first")
resBPD16$name<-mapIds(org.Hs.eg.db, keys=row.names(resBPD16), column="GENENAME", keytype="ENSEMBL", multiVales="first")
resBPD16$entrez<-mapIds(org.Hs.eg.db, keys=row.names(resBPD16), column="ENTREZID", keytype="ENSEMBL", multiVales="first")

#BPD24
resBPD24$symbol<-mapIds(org.Hs.eg.db, keys=row.names(resBPD24), column="SYMBOL", keytype="ENSEMBL", multiVales="first")
resBPD24$name<-mapIds(org.Hs.eg.db, keys=row.names(resBPD24), column="GENENAME", keytype="ENSEMBL", multiVales="first")
resBPD24$entrez<-mapIds(org.Hs.eg.db, keys=row.names(resBPD24), column="ENTREZID", keytype="ENSEMBL", multiVales="first")

#TRP16
resTRP16$symbol<-mapIds(org.Hs.eg.db, keys=row.names(resTRP16), column="SYMBOL", keytype="ENSEMBL", multiVales="first")
resTRP16$name<-mapIds(org.Hs.eg.db, keys=row.names(resTRP16), column="GENENAME", keytype="ENSEMBL", multiVales="first")
resTRP16$entrez<-mapIds(org.Hs.eg.db, keys=row.names(resTRP16), column="ENTREZID", keytype="ENSEMBL", multiVales="first")

#TRP24
resTRP24$symbol<-mapIds(org.Hs.eg.db, keys=row.names(resTRP24), column="SYMBOL", keytype="ENSEMBL", multiVales="first")
resTRP24$name<-mapIds(org.Hs.eg.db, keys=row.names(resTRP24), column="GENENAME", keytype="ENSEMBL", multiVales="first")
resTRP24$entrez<-mapIds(org.Hs.eg.db, keys=row.names(resTRP24), column="ENTREZID", keytype="ENSEMBL", multiVales="first")

#Mock ie UTD24
resMock$symbol<-mapIds(org.Hs.eg.db, keys=row.names(resMock), column="SYMBOL", keytype="ENSEMBL", multiVales="first")
resMock$name<-mapIds(org.Hs.eg.db, keys=row.names(resMock), column="GENENAME", keytype="ENSEMBL", multiVales="first")
resMock$entrez<-mapIds(org.Hs.eg.db, keys=row.names(resMock), column="ENTREZID", keytype="ENSEMBL", multiVales="first")

#write relevant comparisons to csv
write.csv(as.data.frame(resBPD16), file="HeLa_BPD16_DESeq2_lfc_04042021.csv")
write.csv(as.data.frame(resBPD24), file="HeLa_BPD24_DESeq2_lfc_04042021.csv")
write.csv(as.data.frame(resTRP16), file="HeLa_TRP16_DESeq2_lfc_04042021.csv")
write.csv(as.data.frame(resTRP24), file="HeLa_TRP24_DESeq2_lfc_04042021.csv")
write.csv(as.data.frame(resMock), file="HeLa_Mock_DESeq2_lfc_04042021.csv")

#plot PCA
rld<-rlog(dds, blind=FALSE)
pca<-plotPCA(rld, intgroup="condition", returnData=TRUE)
pVar<-round(100*attr(pca, "percentVar"))
ggplot(pca, aes(PC1, PC2, color=condition)) +
    geom_point(size=3) +
    xlab(paste0("PC1: ",pVar[1],"% variance")) +
    ylab(paste0("PC2: ",pVar[2],"% variance")) +
    coord_fixed() +
    theme_linedraw() +
    theme(legend.title=element_blank()) +
    ggtitle("HeLa PCA") +
    theme(plot.title = element_text(hjust=0.5)) +
    geom_text(aes(label = name), hjust=0, vjust=0) +
    xlim(-55,45)

#generate MA plot
plotMA(resBPD16, main="BPD16\nHeLa", ylim=c(-6,6))
plotMA(resBPD24, main="BPD24\nHeLa", ylim=c(-6,6))
plotMA(resTRP16, main="TRP16\nHeLa", ylim=c(-6,6))
plotMA(resTRP24, main="TRP24\nHeLa", ylim=c(-6,6))
plotMA(resMock, main="Mock\nHeLa", ylim=c(-6,6))

#extract Normalized counts
dds <- estimateSizeFactors(dds)
ncts<-counts(dds, normalized=TRUE)
write.csv(ncts, file="HeLa_NormCounts_DESeq2_04042021.csv")

#EnhancedVolcano Plots
#UTD24
UTD24_vol<-read.csv("HeLa_UTD24_DESeq2_lfc_04042021.csv")
UTD24_vol$padj2 <- ifelse(UTD24_vol$padj < 1e-50, 1e-50, UTD24_vol$padj)
EnhancedVolcano(UTD24_vol, lab=UTD24_vol$symbol, x='log2FoldChange', y='padj2', pCutoff = 5e-2, legendPosition = '', title="UTD24", subtitle="Reference = Mock") +
    theme(plot.title = element_text(hjust = 0.5)) +
    theme(plot.subtitle = element_text(hjust = 0.5)) +
    coord_cartesian(ylim = c(-2, 50))

#adjust ylim
EnhancedVolcano(UTD24_vol, lab=UTD24_vol$symbol, x='log2FoldChange', y='padj', pCutoff = 5e-2, legendPosition = '', title="UTD24", subtitle="Reference = Mock") + theme(plot.title = element_text(hjust = 0.5)) + theme(plot.subtitle = element_text(hjust = 0.5)) + coord_cartesian(ylim = c(-2.5, 65))

#BPD16
BPD16_vol<-read.csv("HeLa_BPD16_DESeq2_lfc_04042021.csv")
BPD16_vol$padj2 <- ifelse(BPD16_vol$padj < 1e-30, 1e-30, BPD16_vol$padj)
EnhancedVolcano(BPD16_vol, lab=BPD16_vol$symbol, x='log2FoldChange', y='padj2', pCutoff = 5e-2, legendPosition = '', title="BPD16", subtitle="Reference = UTD24") +
    theme(plot.title = element_text(hjust = 0.5)) +
    theme(plot.subtitle = element_text(hjust = 0.5)) +
    coord_cartesian(ylim = c(-2, 30))

#adjust ylim
EnhancedVolcano(BPD16_vol, lab=BPD16_vol$symbol, x='log2FoldChange', y='padj', pCutoff = 5e-2, legendPosition = '', title="BPD16", subtitle="Reference = UTD24") + theme(plot.title = element_text(hjust = 0.5)) + theme(plot.subtitle = element_text(hjust = 0.5)) + coord_cartesian(ylim = c(-2.5, 60)) 

#TRP16
TRP16_vol<-read.csv("HeLa_TRP16_DESeq2_lfc_04042021.csv")
TRP16_vol$padj2 <- ifelse(TRP16_vol$padj < 1e-30, 1e-30, TRP16_vol$padj)
EnhancedVolcano(TRP16_vol, lab=TRP16_vol$symbol, x='log2FoldChange', y='padj2', pCutoff = 5e-2, legendPosition = '', title="TRP16", subtitle="Reference = UTD24") +
    theme(plot.title = element_text(hjust = 0.5)) +
    theme(plot.subtitle = element_text(hjust = 0.5)) +
    coord_cartesian(ylim = c(-2, 30))

#BPD24
BPD24_vol<-read.csv("HeLa_BPD24_DESeq2_lfc_04042021.csv")
BPD24_vol$padj2 <- ifelse(BPD24_vol$padj < 1e-50, 1e-50, BPD24_vol$padj)
EnhancedVolcano(BPD24_vol, lab=BPD24_vol$symbol, x='log2FoldChange', y='padj2', pCutoff = 5e-2, legendPosition = '', title="BPD24", subtitle="Reference = UTD24") +
    theme(plot.title = element_text(hjust = 0.5)) +
    theme(plot.subtitle = element_text(hjust = 0.5)) +
    coord_cartesian(ylim = c(-2, 50))

#adjust ylim
EnhancedVolcano(BPD24_vol, lab=BPD24_vol$symbol, x='log2FoldChange', y='padj', pCutoff = 5e-2, legendPosition = '', title="BPD24", subtitle="Reference = UTD24") + theme(plot.title = element_text(hjust = 0.5)) + theme(plot.subtitle = element_text(hjust = 0.5)) + coord_cartesian(ylim = c(-2.5, 100)) 

#TRP24
TRP24_vol<-read.csv("HeLa_TRP24_DESeq2_lfc_04042021.csv")
TRP24_vol$padj2 <- ifelse(TRP24_vol$padj < 1e-30, 1e-30, TRP24_vol$padj)
EnhancedVolcano(TRP24_vol, lab=TRP24_vol$symbol, x='log2FoldChange', y='padj2', pCutoff = 5e-2, legendPosition = '', title="TRP24", subtitle="Reference = UTD24") +
    theme(plot.title = element_text(hjust = 0.5)) +
    theme(plot.subtitle = element_text(hjust = 0.5)) +
    coord_cartesian(ylim = c(-2, 30))

#clusterProfiler analysis
 if (!requireNamespace("BiocManager", quietly = TRUE))
+     install.packages("BiocManager")
 
BiocManager::install("clusterProfiler")
install.packages('ggnewscale')
install.packages('ggridges')
install.packages('clusterProfiler.dplyr')
library(ggnewscale)
library(ggridges)
library(DOSE)
library(ggplot2)
library(clusterProfiler)
library(dplyr)
library(org.Hs.eg.db)
library(enrichplot)
library(GOSemSim)

#UTD24
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/HeLa/Final")
#read DESeq2 file
UTD24<-read.csv("HeLa_UTD24_DESeq2_lfc_04042021.csv")

#remove all duplicated entrez IDs
UTD24<-UTD24%>%distinct(entrez, .keep_all = TRUE)
#remove all NA values
UTD24 <- UTD24[complete.cases(UTD24),]
#extract log2FoldChange column
UTD24list <- UTD24[,3]
#assign entrez IDs as row names
names(UTD24list) <- as.character(UTD24[,9])
#sort log2FoldChange in decreasing order
UTD24list <- sort(UTD24list, decreasing = TRUE)
#performe gseKEGG analysis
UTD24k <- gseKEGG(geneList = UTD24list, organism = 'hsa', minGSSize = 100, pvalueCutoff = 0.01, verbose = FALSE, keyType = 'kegg')
#write data as csv file
write.csv(as.data.frame(UTD24k), file="HeLa_UTD24_gseKEGG_04142021.csv")

#dotplot of top 10 activated and suppressed pathways
dotplot(UTD24k, showCategory=10, orderBy = "x", split=".sign") +
  ggtitle("UTD24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  facet_grid(.~.sign) +
  scale_x_continuous(limits=c(0.1, 0.4))

#mutate gseKEGG data to sort NES in descending order, i.e. most positively-enriched pathways are at top of list
mUTD24k <- mutate(UTD24k) %>% arrange(desc(NES))
#write this data as csv file
write.csv(as.data.frame(mUTD24k), file="HeLa_UTD24_arrangeNES_gseKEGG_04142021.csv")
#dotplot of top 15 enriched pathways
dotplot(mUTD24k, showCategory = 15, orderBy = "x") +
  ggtitle("UTD24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_x_continuous(limits=c(0.1, 0.4))

#compute similarity between KEGG pathway terms
pUTD24k <- pairwise_termsim(mUTD24k, method = "JC", semData = NULL, showCategory = 400)
#emapplot of similarity between top 15 enriched KEGG terms
emapplot(pUTD24k, showCategory = 15)

#ridgeline plot of top 15 significant KEGG terms
ridgeplot(UTD24k, showCategory = 15) +
  ggtitle("UTD24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab("Log2 Fold Change")

UTD24l<-read.csv("HeLa_UTD24_gseKEGG_04142021.csv")
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/HeLa/Final/clusterProfiler/gseKEGG/UTD24")
tmp = sapply(UTD24l, function(pid) pathview(gene.data=UTD24list, gene.idtype="kegg", pathway.id=pid, species="hsa", same.layer = F, low="purple", mid="gray", high="yellow"))
#purine metabolism
tmp = sapply(UTD24l, function(pid) pathview(gene.data=UTD24list, gene.idtype="kegg", pathway.id='hsa00230', species="hsa", same.layer = F, low="purple", mid="gray", high="yellow"))

#BPD16
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/HeLa/Final")
BPD16<-read.csv("HeLa_BPD16_DESeq2_lfc_04042021.csv")

BPD16<-BPD16%>%distinct(entrez, .keep_all = TRUE)
BPD16 <- BPD16[complete.cases(BPD16),]
BPD16list <- BPD16[,3]
names(BPD16list) <- as.character(BPD16[,9])
BPD16list <- sort(BPD16list, decreasing = TRUE)
BPD16k <- gseKEGG(geneList = BPD16list, organism = 'hsa', minGSSize = 100, pvalueCutoff = 0.01, verbose = FALSE, keyType = 'kegg')
write.csv(as.data.frame(BPD16k), file="HeLa_BPD16_gseKEGG_04142021.csv")

dotplot(BPD16k, showCategory=10, orderBy = "x", split=".sign") +
  ggtitle("BPD16 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  facet_grid(.~.sign) +
  scale_x_continuous(limits=c(0.1, 0.7))

mBPD16k <- mutate(BPD16k) %>% arrange(desc(NES))
write.csv(as.data.frame(mBPD16k), file="HeLa_BPD16_arrangeNES_gseKEGG_04142021.csv")
dotplot(mBPD16k, showCategory = 15, orderBy = "x") +
  ggtitle("BPD16 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_x_continuous(limits=c(0.1, 0.7))

pBPD16k <- pairwise_termsim(mBPD16k, method = "JC", semData = NULL, showCategory = 400)
emapplot(pBPD16k, showCategory = 15)

ridgeplot(BPD16k, showCategory = 15) +
  ggtitle("BPD16 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab("Log2 Fold Change")

BPD16l<-read.csv("HeLa_BPD16_gseKEGG_04142021.csv")
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/HeLa/Final/clusterProfiler/gseKEGG/BPD16")
tmp = sapply(BPD16l, function(pid) pathview(gene.data=BPD16list, gene.idtype="kegg", pathway.id=pid, species="hsa", same.layer = F, low="purple", mid="gray", high="yellow"))
#purine metabolism
tmp = sapply(BPD16l, function(pid) pathview(gene.data=BPD16list, gene.idtype="kegg", pathway.id='hsa00230', species="hsa", same.layer = F, low="purple", mid="gray", high="yellow"))

#TRP16
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/HeLa/Final")
TRP16<-read.csv("HeLa_TRP16_DESeq2_lfc_04042021.csv")

TRP16<-TRP16%>%distinct(entrez, .keep_all = TRUE)
TRP16 <- TRP16[complete.cases(TRP16),]
TRP16list <- TRP16[,3]
names(TRP16list) <- as.character(TRP16[,9])
TRP16list <- sort(TRP16list, decreasing = TRUE)
TRP16k <- gseKEGG(geneList = TRP16list, organism = 'hsa', minGSSize = 100, pvalueCutoff = 0.01, verbose = FALSE, keyType = 'kegg')
write.csv(as.data.frame(TRP16k), file="HeLa_TRP16_gseKEGG_04142021.csv")

mTRP16k <- mutate(TRP16k) %>% arrange(desc(NES))
write.csv(as.data.frame(mTRP16k), file="HeLa_TRP16_arrangeNES_gseKEGG_04142021.csv")
dotplot(mTRP16k, showCategory = 15, orderBy = "x") +
  ggtitle("TRP16 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_x_continuous(limits=c(0.1, 0.4))

pTRP16k <- pairwise_termsim(mTRP16k, method = "JC", semData = NULL, showCategory = 400)
emapplot(pTRP16k, showCategory = 15)

ridgeplot(TRP16k, showCategory = 15) +
  ggtitle("TRP16 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab("Log2 Fold Change")

TRP16l<-read.csv("HeLa_TRP16_gseKEGG_04142021.csv")
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/HeLa/Final/clusterProfiler/gseKEGG/TRP16")
tmp = sapply(TRP16l, function(pid) pathview(gene.data=TRP16list, gene.idtype="kegg", pathway.id=pid, species="hsa", same.layer = F, low="purple", mid="gray", high="yellow"))
#purine metabolism
tmp = sapply(TRP16l, function(pid) pathview(gene.data=TRP16list, gene.idtype="kegg", pathway.id='hsa00230', species="hsa", same.layer = F, low="purple", mid="gray", high="yellow"))

#BPD24
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/HeLa/Final")
BPD24<-read.csv("HeLa_BPD24_DESeq2_lfc_04042021.csv")

BPD24<-BPD24%>%distinct(entrez, .keep_all = TRUE)
BPD24 <- BPD24[complete.cases(BPD24),]
BPD24list <- BPD24[,3]
names(BPD24list) <- as.character(BPD24[,9])
BPD24list <- sort(BPD24list, decreasing = TRUE)
BPD24k <- gseKEGG(geneList = BPD24list, organism = 'hsa', minGSSize = 100, pvalueCutoff = 0.01, verbose = FALSE, keyType = 'kegg')
write.csv(as.data.frame(BPD24k), file="HeLa_BPD24_gseKEGG_04142021.csv")

dotplot(BPD24k, showCategory=10, orderBy = "x", split=".sign") +
  ggtitle("BPD24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  facet_grid(.~.sign) +
  scale_x_continuous(limits=c(0.1, 0.7))

mBPD24k <- mutate(BPD24k) %>% arrange(desc(NES))
write.csv(as.data.frame(mBPD24k), file="HeLa_BPD24_arrangeNES_gseKEGG_04142021.csv")
dotplot(mBPD24k, showCategory = 15, orderBy = "x") +
  ggtitle("BPD24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_x_continuous(limits=c(0.1, 0.7))

pBPD24k <- pairwise_termsim(mBPD24k, method = "JC", semData = NULL, showCategory = 400)
emapplot(pBPD24k, showCategory = 15)

ridgeplot(BPD24k, showCategory = 15) +
  ggtitle("BPD24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab("Log2 Fold Change")

BPD24l<-read.csv("HeLa_BPD24_gseKEGG_04142021.csv")
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/HeLa/Final/clusterProfiler/gseKEGG/BPD24")
tmp = sapply(BPD24l, function(pid) pathview(gene.data=BPD24list, gene.idtype="kegg", pathway.id=pid, species="hsa", same.layer = F, low="purple", mid="gray", high="yellow"))
#purine metabolism
tmp = sapply(BPD24l, function(pid) pathview(gene.data=BPD24list, gene.idtype="kegg", pathway.id='hsa00230', species="hsa", same.layer = F, low="purple", mid="gray", high="yellow"))

#TRP24
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/HeLa/Final")
TRP24<-read.csv("HeLa_TRP24_DESeq2_lfc_04042021.csv")

TRP24<-TRP24%>%distinct(entrez, .keep_all = TRUE)
TRP24 <- TRP24[complete.cases(TRP24),]
TRP24list <- TRP24[,3]
names(TRP24list) <- as.character(TRP24[,9])
TRP24list <- sort(TRP24list, decreasing = TRUE)
TRP24k <- gseKEGG(geneList = TRP24list, organism = 'hsa', minGSSize = 100, pvalueCutoff = 0.01, verbose = FALSE, keyType = 'kegg')
write.csv(as.data.frame(TRP24k), file="HeLa_TRP24_gseKEGG_04142021.csv")

mTRP24k <- mutate(TRP24k) %>% arrange(desc(NES))
write.csv(as.data.frame(mTRP24k), file="HeLa_TRP24_arrangeNES_gseKEGG_04142021.csv")
dotplot(mTRP24k, showCategory = 15, orderBy = "x") +
  ggtitle("TRP24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_x_continuous(limits=c(0.1, 0.5))

pTRP24k <- pairwise_termsim(mTRP24k, method = "JC", semData = NULL, showCategory = 400)
emapplot(pTRP24k, showCategory = 15)

ridgeplot(TRP24k, showCategory = 15) +
  ggtitle("TRP24 KEGG") +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab("Log2 Fold Change")

TRP24l<-read.csv("HeLa_TRP24_gseKEGG_04142021.csv")
setwd("~/Documents/Carabeo Lab/R Code/Dual RNA-Seq/DESeq2/HeLa/Final/clusterProfiler/gseKEGG/TRP24")
tmp = sapply(TRP24l, function(pid) pathview(gene.data=TRP24list, gene.idtype="kegg", pathway.id=pid, species="hsa", same.layer = F, low="purple", mid="gray", high="yellow"))
#purine metabolism
tmp = sapply(TRP24l, function(pid) pathview(gene.data=TRP24list, gene.idtype="kegg", pathway.id='hsa00230', species="hsa", same.layer = F, low="purple", mid="gray", high="yellow"))
```
